src/test/java/com/exasol/projectkeeper/repository/GitRepositoryTest.java

```diff
@@ -0,0 +1,126 @@
+            final GitRepository repository = new GitRepository(this.tempDir);
+            final List<TaggedCommit> result = repository.getTagsInCurrentBranch();
+            final List<String> tagNames = result.stream().map(TaggedCommit::getTag).collect(Collectors.toList());
+            assertThat(tagNames, contains("tag1"));
+        }
+    }
+
+    @Test
+    void testGetTagsFromDetachedHead() throws GitAPIException, IOException {
+        try (final Git git = Git.init().setDirectory(this.tempDir.toFile()).call();) {
+            git.checkout().setName("main").setOrphan(true).call();
+            makeCommitAndTag(git, 1, true);
+            final RevCommit secondCommit = makeCommitAndTag(git, 2, true);
+            git.checkout().setName(secondCommit.getName()).call();
+            final GitRepository repository = new GitRepository(this.tempDir);
+            final List<TaggedCommit> result = repository.getTagsInCurrentBranch();
+            final List<String> tagNames = result.stream().map(TaggedCommit::getTag).collect(Collectors.toList());
+            assertThat(tagNames, contains("tag2", "tag1"));
+        }
+    }
+
+    @Test
+    void testGetTagsInNoGitDirectory() {
+        final GitRepository repository = new GitRepository(this.tempDir);
+        final IllegalStateException exception = assertThrows(IllegalStateException.class,
+                repository::getTagsInCurrentBranch);
+        assertThat(exception.getMessage(), startsWith("E-PK-32"));
+    }
+
+    private RevCommit makeCommitAndTag(final Git git, final int counter, final boolean lightweight)
+            throws IOException, GitAPIException {
+        Files.writeString(this.tempDir.resolve("myFile.txt"), counter + "");
+        git.add().addFilepattern("myFile.txt").call();
+        final RevCommit commit = git.commit().setMessage(counter + ". commit").call();
+        git.tag().setName("tag" + counter).setAnnotated(!lightweight).call();
+        return commit;
+    }
+
+    @Test
+    void testReadFileAtCommit() throws GitAPIException, IOException {
+        try (final Git git = Git.init().setDirectory(this.tempDir.toFile()).call()) {
+            final Path testFile = this.tempDir.resolve("myFile.txt");
+            final String fileContent = "some string";
+            Files.writeString(testFile, fileContent);
+            git.add().addFilepattern("myFile.txt").call();
+            final RevCommit initialCommit = git.commit().setMessage("initial commit").call();
+            Files.delete(testFile);
+            git.add().addFilepattern("myFile.txt").call();
+            git.commit().setMessage("removed file").call();
+
+            final GitRepository repository = new GitRepository(this.tempDir);
+            final String result = repository.readFileAtCommit(Path.of("myFile.txt"), new GitCommit(initialCommit));
+            assertThat(result, equalTo(fileContent));
+        }
+    }
+
+    @Test
+    void testReadFileAtCommitInNoGitDirectory() {
```

Comment:
I have a feeling that you're testing third-party library in this class

Reply:
well yes, but mocking the git part does not look like a good option to me. These tests make at least sure that the class behaves as expected.

Reply:
The tests here are integration tests, right? Maybe we can rename the class to say explicitly that it's integration test?